test/units/module_utils/common/text/test_converters.py

```diff
@@ -0,0 +1,240 @@
+            ('str', b'str'),
+            (['str'], [b'str']),
+            (('str'), (b'str')),
+            ({'str': 'str'}, {b'str': b'str'}),
+        ]
+    )
+    @pytest.mark.parametrize('encoding', ['utf-8', 'ascii'])
+    @pytest.mark.parametrize('errors', ['strict', 'ignore'])
+    def test_to_bytes_different_types(self, test_input, expected, encoding, errors):
+        """Test for passing objects to container_to_bytes()."""
+
+        assert container_to_bytes(test_input, encoding=encoding, errors=errors) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (True, True),
+            (None, None),
+            ('str', 'str'.encode(DEFAULT_ENCODING)),
+            (['str'], ['str'.encode(DEFAULT_ENCODING)]),
+            (('str'), ('str'.encode(DEFAULT_ENCODING))),
+            ({'str': 'str'}, {'str'.encode(DEFAULT_ENCODING): 'str'.encode(DEFAULT_ENCODING)}),
+        ]
+    )
+    def test_to_bytes_default_encoding_err(self, test_input, expected):
+        """
+        Test for passing objects to container_to_bytes(). Default encoding and errors
+        """
+
+        assert container_to_bytes(test_input) == expected
+
+
+class Test_container_to_text:
+
+    """Namespace for testing container_to_text()."""
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (b'str', 'str'),
+            ([b'str'], ['str']),
+            ((b'str'), ('str')),
+            ({b'str': b'str'}, {'str': 'str'}),
+        ]
+    )
+    @pytest.mark.parametrize('encoding', ['utf-8', 'ascii'])
+    @pytest.mark.parametrize('errors', ['strict', 'ignore'])
+    def test_to_text_different_types(self, test_input, expected, encoding, errors):
+        """Test for passing objects to container_to_text()."""
+
+        assert container_to_text(test_input, encoding=encoding, errors=errors) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (True, True),
+            (None, None),
+            ('str'.encode(DEFAULT_ENCODING), 'str'),
+            (['str'.encode(DEFAULT_ENCODING)], ['str']),
+            (('str'.encode(DEFAULT_ENCODING)), ('str')),
+            ({'str'.encode(DEFAULT_ENCODING): 'str'.encode(DEFAULT_ENCODING)}, {'str': 'str'}),
+        ]
+    )
+    def test_to_text_default_encoding_err(self, test_input, expected):
+        """
+        Test for passing objects to container_to_text(). Default encoding and errors
+        """
+
+        assert container_to_text(test_input) == expected
+
+
+class Test_jsonify:
```

Comment:
Having a class for just one test method is unnecessary.

Reply:
It's a good way to separate code (in case of pytest, as just a namespace)

Reply:
(BTW, I've seen the video you shared in irc about classes and functions on pycon - good job! However in this particular case it is just a namespace, easier to work with them when tests are separate)